chainer/functions/pooling/roi_align_2d.py

```diff
@@ -0,0 +1,554 @@
+# Modified work:
+# -----------------------------------------------------------------------------
+# Copyright (c) 2018 Preferred Infrastructure, Inc.
+# Copyright (c) 2018 Preferred Networks, Inc.
+# -----------------------------------------------------------------------------
+
+# Original work:
+# -----------------------------------------------------------------------------
+# Copyright (c) 2015 by Contributors
+# \file roi_pooling.cu
+# \brief roi pooling operator
+# \author Ross Girshick, Kye-Hyeon Kim, Jian Guo
+# \changed to roi_align by Elaine Bao
+# \file roi_align.cu
+# \roi align operator described in Mask RCNN
+# -----------------------------------------------------------------------------
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class ROIAlign2D(function.Function):
+
+    """ROI align over a set of 2d planes."""
+
+    def __init__(self, outh, outw, spatial_scale, sampling_ratio=0):
+        for arg in ['outh', 'outw', 'sampling_ratio']:
+            value = eval(arg)
+            if not (isinstance(value, int) and value >= 0):
+                raise TypeError(
+                    '{} must be positive integer: {}, {}'
+                    .format(arg, type(value), value)
+                )
+        if isinstance(spatial_scale, int):
+            spatial_scale = float(spatial_scale)
+        elif not isinstance(spatial_scale, float):
+            raise TypeError(
+                'spatial_scale must be float: {}'
+                .format(type(spatial_scale), spatial_scale)
+            )
```

Comment:
Can `spatial_scale` be negative? Could you add another check to ensure `spatial_scale` is positive?